apps/test-suite/stories/ui-explorer/insertBetween.js

```diff
@@ -0,0 +1,9 @@
+const insertBetween = (item, array) =>
+  array.reduce((acc, curr, i, { length }) => {
```

Comment:
reduce the memory copying and over-abstraction

```js
const result = [];
for (const i = 0; i < array.length; i++) {
  result.push(array[i]);
  if (i !== array.length) {
    result.push(item());
  }
}
return result;
```

Reply:
The `ui-explorer/` part is generic, perhaps someone has rolled it into a npm package.